lib/ecto/adapters/postgres/connection.ex

```diff
@@ -409,8 +409,9 @@ if Code.ensure_loaded?(Postgrex) do
 
     defp cte(%{with_ctes: _}, _), do: []
 
-    defp cte_expr({name, cte}, sources, query) do
-      [quote_name(name), " AS ", cte_query(cte, sources, query)]
+    defp cte_expr({name, materialize, cte}, sources, query) do
+      materialize_opt = if materialize, do: "MATERIALIZED ", else: ""
+      [quote_name(name), " AS ", materialize_opt, cte_query(cte, sources, query)]
     end
```

Comment:
If this gets accepted, I'll add support for this to `ecto_sqlite3` it supports materialized CTEs https://www.sqlite.org/lang_with.html